src/python/pants/goal/help.py

```diff
@@ -73,31 +74,30 @@ def create_help_request(self, options: Options) -> HelpRequest:
         return NoGoalHelp()
 
 
-class ThingHelpBuiltinGoal(HelpBuiltinGoalBase):
-    name = "help"
-    help = "Display usage message."
-    aliases = (
-        "-h",
-        "--help",
-    )
+class ThingHelpBuiltinGoalBase(HelpBuiltinGoalBase):
```

Comment:
Refactored to avoid code duplication of the `create_help_request` method for thing (advanced) help.